tools/k8s_versions/update_k8s_versions.go

```diff
@@ -0,0 +1,222 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type KubernetesVersion struct {
+	Cycle       string `json:"cycle"`
+	ReleaseDate string `json:"releaseDate"`
+	EOLDate     string `json:"eol"`
+	Latest      string `json:"latest"`
+}
+
+type DockerImage struct {
+	Count int `json:"count"`
+}
+
+func fetchKubernetesVersions(url string) ([]KubernetesVersion, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting k8s versions: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	var kubernetesVersions []KubernetesVersion
+	if err := json.Unmarshal(body, &kubernetesVersions); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal JSON: %w", err)
+	}
+
+	return kubernetesVersions, nil
+}
+
+// yaml parsing does not preserve the original format, mainly blank lines.
+// This function reads the raw yaml file, locate and update the k8s-version block
+func updateK8sVersion(filePath string, kubernetesVersions []KubernetesVersion) error {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return fmt.Errorf("failed to open file: %w", err)
+	}
+	defer file.Close()
+
+	var output []string
+	var k8sVersionBlock []string
+	var inK8sVersionBlock bool
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		trimmed := strings.TrimSpace(line)
+
+		// Locate the k8s-version block needed to update
+		if strings.HasPrefix(trimmed, "k8s-version:") || strings.HasPrefix(trimmed, "kubernetes_version:") {
+			inK8sVersionBlock = true
+			output = append(output, line)
+			continue
+		}
+
+		if inK8sVersionBlock {
+			if strings.HasPrefix(trimmed, "-") || strings.HasPrefix(trimmed, "#") {
+				k8sVersionBlock = append(k8sVersionBlock, line)
```

Comment:
nit: a comment describing that here the code is collecting all entries in the list of version or some to that effect